### PR TITLE
Add "Exit Code Editor" toolbar.

### LINKED ISSUE #9445: Once in Code Editor provide suggestion on how to get back to visual editor
If you click into the Code Editor you are brought t an editor with a number of comments and a bunch of HTML code. If you didn't know where you click or what you clicked you / a basic WordPress user will be very confused as to what you did to find yourself in the code editor and how to get back. Since the switch code editor / visual editor is hidden behind a hamburger style menu of three dots and a user would go searching for how to get back to the visual editor. 

If you try to click undo you are actually undoing changes you made to the page, not undoing clicking to go to the code editor. 

We provided warnings in [WordPress 4.9 that you were going to the code editor](https://make.wordpress.org/core/2017/10/22/code-editing-improvements-in-wordpress-4-9/) I believe and lobby for at least an easy way to get back to the visual editor (just the first time) or we will have a lot of lost users editing code they shouldn't be wondering how they will get back and clicking everywhere to do so.

**Expected behavior**
If you want the switch between the code and visual editor hidden then A line of text with a yellow background saying that you have gone into the code editor and asking if that is really where you wanted to go, with a text link back to the visual editor.  

**Suggested notification**
Easy Suggested Notification to the user who has entered the code editor.

screen shot 2018-08-29 at 1 45 55 pm

**Screenshots**
screen shot 2018-08-29 at 12 24 52 pm

**Desktop**
 - OS: OSX 10.11.6
 - Browser Firefox 
 - Version 61

### PR DESCRIPTION
**Pull request by jasmussen**
This PR intends to fix #9445. It adds a toolbar to the code editor with a label showing what you are doing, as well as an "Exit Code Editor" button that takes you back to the visual editor.

I hope the space between this label and the close button can be used, in the future, to add a slot fill so a plugin could add quicktags here.

Screenshot:

screen shot 2018-09-20 at 11 08 43

The PR also cleans up a few things, removes some redundant rules, refactors the centering stuff. It is tested for all breakpoints. However, it is not yet wired up, and I could use some help here. Essentially the button doesn't do anything yet. @tofumatt or @gziolo can you help me wire this up? Thanks.

### CODE REVIEW

**Comment by jasmussen** [comment]
Thanks, I don't think a fix is urgent. Would love your help some time next week, thank you.

> First thoughts though is the "Exit Code Editor" thing doesn't look super "button-y"… any way to make it seem more like something I'd expect to click on?

It's buttony as soon as you hover :D and in that vein it follows the same pattern as the "Close" button on the sidebar.

Adding "Needs Design Feedback" for thoughts on that! Thanks for your help!

**Comment by gziolo** [comment]
Switching back to the visual mode is super slow. We probably should optimize it in a separate PR. From what I see we unmount visual editor when going into code editing mode. In my opinion, we should have visual mode always rendered but hidden with CSS when we switch to the text mode to improve the user experience and accommodate. Another option is to finally add some optimizations to the editor's visual mode to render only those blocks that are visible on the screen. /cc @mtias

**Review by gziolo — APPROVED** [review]
LGTM after my commit landed 💯

**Comment by jasmussen** [comment]
Thank you so much for the help!

**Comment by mtias** [comment]
@gziolo could also try implementing the Editor Mode as a modal.

**Comment by mtias** [comment]
The modal would also have the advantage of having a clear non-ad-hoc closing mechanism.

**Comment by gziolo** [comment]
That's an interesting idea. Initially, I thought about making it look like a screen takeover but we don't have this UI piece yet :) Modal seems like a good compromise. There is still a question what about those users which prefer to use Code editor instead as a default mode. It was raised by @jasmussen in our chat on Slack.

**Comment by jasmussen** [comment]
Either way I'm very glad to have this clear affordance in place, as it fixes a usability issue where it can be hard to switch back. Separate improvements would be lovely to explore in separate PRs. Thanks again!

**Pull request merged**
